clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1171,4 +1253,43 @@ private void release() {
         if (refcount.decrementAndGet() == 0)
             currentThread.set(NO_CURRENT_THREAD);
     }
+
+    private class PatternSubscriptionTask implements DelayedTask {
+        private final long interval;
+
+        public PatternSubscriptionTask(long interval) {
+            this.interval = interval;
+        }
+
+        public void run(final long now) {
+            final List<String> topicsToSubscribe = new ArrayList<>();
+
+            final RequestFuture<ClientResponse> requestFuture = fetcher.sendMetadataRequest();
+            if (requestFuture != null) {
+                requestFuture.addListener(new RequestFutureListener<ClientResponse>() {
+                    @Override
+                    public void onSuccess(ClientResponse value) {
+                        MetadataResponse response =
```

Comment:
@SinghAsDev Haha, I can understand why this is a little confusing, but the network layer considers the request a "success" if it gets a response. However, that doesn't mean that there wasn't an error code in that response. It might be nice to handle this at the network layer, but it seems to me that there wasn't a generic way to check for errors. Each response object had the error code at a different location in its schema, so the only thing we could do is pass the response back and let the application determine if there was an error. The one case where we might be able to infer errors generically is by checking ClientResponse.wasDisconnected(), but I don't think we do this either (I've forgotten if there's a good reason for that).